ublox_derive/src/input.rs

```diff
@@ -0,0 +1,643 @@
+        }
+        return Err(Error::new(
+            ast.repr_ty.span(),
+            format!("Not supported type, expect one of {:?}", valid_type_names),
+        ));
+    };
+
+    let mut consts = Vec::with_capacity(ast.items.len());
+    for item in ast.items {
+        consts.push(BitFlagsMacroItem {
+            attrs: item.attrs,
+            name: item.name,
+            value: item.value.base10_parse()?,
+        });
+    }
+
+    Ok(BitFlagsMacro {
+        nbits,
+        vis: ast.vis,
+        attrs: ast.attrs,
+        name: ast.ident,
+        repr_ty: ast.repr_ty,
+        consts,
+        from_fn,
+        into_fn,
+        rest_handling,
+    })
+}
+
+pub fn parse_idents_list(input: proc_macro2::TokenStream) -> syn::Result<RecvPackets> {
+    syn::parse2(input)
+}
+
+fn parse_ubx_extend_attrs(
+    ubx_extend_name: &str,
+    item_sp: Span,
+    attrs: &[Attribute],
+) -> syn::Result<(
+    Option<UbxTypeFromFn>,
+    Option<UbxTypeIntoFn>,
+    Option<UbxEnumRestHandling>,
+)> {
+    let attr = attrs
+        .iter()
+        .find(|a| a.path.is_ident("ubx"))
+        .ok_or_else(|| Error::new(item_sp, format!("No ubx attribute for {}", ubx_extend_name)))?;
+    let meta = attr.parse_meta()?;
+    let mut from_fn = None;
+    let mut rest_handling = None;
+    let mut into_fn = None;
+    let meta_sp = meta.span();
+    match meta {
+        syn::Meta::List(list) => {
+            for item in list.nested {
+                if let syn::NestedMeta::Meta(syn::Meta::Path(p)) = item {
+                    if p.is_ident("from") {
+                        from_fn = Some(UbxTypeFromFn::From);
+                    } else if p.is_ident("into_raw") {
+                        into_fn = Some(UbxTypeIntoFn::Raw);
+                    } else if p.is_ident("from_unchecked") {
+                        from_fn = Some(UbxTypeFromFn::FromUnchecked);
+                    } else if p.is_ident("rest_reserved") {
+                        rest_handling = Some(UbxEnumRestHandling::Reserved);
+                    } else if p.is_ident("rest_error") {
+                        rest_handling = Some(UbxEnumRestHandling::ErrorProne);
+                    } else {
+                        return Err(syn::Error::new(p.span(), "Invalid ubx attribute"));
+                    }
+                } else {
+                    return Err(syn::Error::new(item.span(), "Invalid ubx attribute"));
+                }
+            }
+        }
+        _ => return Err(syn::Error::new(attr.span(), "Invalid ubx attributes")),
+    }
+
+    if from_fn == Some(UbxTypeFromFn::From)
+        && rest_handling == Some(UbxEnumRestHandling::ErrorProne)
+    {
+        return Err(Error::new(
+            meta_sp,
+            "you should use rest_error with from_unchecked",
+        ));
+    }
+
+    Ok((from_fn, into_fn, rest_handling))
+}
+
+fn parse_ubx_attr(attrs: &[Attribute], struct_name: &Ident) -> syn::Result<PackHeader> {
```

Comment:
`parse_ubx_attr` is only used for the struct parsing. `input.rs` is pretty big, should it be split into different files for interpreting structs and extending enums?

Reply:
I suppose this is very subjective field: right size of module. I would suggest deal with it not in the frame of this PR.